lib/selections/index.js

```diff
@@ -0,0 +1,904 @@
+      }
+    }
+    return result;
+  };
+
+  /**
+   * Return a tuple if the user selected contiguous columns, otherwise `null`.
+   * Info: Because the user may reorder the columns,
+   * the schemaIndex of the first item may be greater than the schemaIndex of the second item,
+   * but the columnIndex of the firs item must less than the columnIndex of the second item.
+   * @param {object[]} [schema] from `self.getSchema()`
+   * @returns {object[]} column schemas tuple (each schema has an additional field `schemaIndex`)
+   */
+  self.getSelectedContiguousColumns = function (schema) {
+    if (!schema) schema = self.getSchema();
+    const columnOrderIndexes = getSelectedContiguousColumns(
+      self.selections,
+      false,
+    );
+    if (!Array.isArray(columnOrderIndexes)) return;
+
+    /** @type {number[]} orders[index] => columnIndex */
+    const orders = self.orders.columns;
+    return columnOrderIndexes.map((orderIndex) => {
+      const schemaIndex = orders[orderIndex];
+      const columnSchema = schema[schemaIndex];
+      return Object.assign({}, columnSchema, { orderIndex });
+    });
+  };
+
+  /**
+   * @param {boolean} [allowOnlyOneRow]
+   * @returns {number[]} a viewRowIndex tuple. It can contains one row index or two row indexes.
+   */
+  self.getSelectedContiguousRows = function (allowOnlyOneRow) {
+    const viewRowIndexes = getSelectedContiguousRows(self.selections, false);
+    if (!Array.isArray(viewRowIndexes)) return;
+    if (!allowOnlyOneRow && viewRowIndexes[0] === viewRowIndexes[1]) return;
+    return viewRowIndexes;
+  };
+
+  self.canSelectionsBeCopied = function () {
+    return self.selections.length > 0 && !areSelectionsComplex(self.selections);
+  };
+
+  self.getSelectedRowsHeight = function () {
+    const selected = getSelectedContiguousRows(self.selections, true);
+    if (!selected) return 0;
+    let height = 0;
+    for (let row = selected[0]; row <= selected[1]; row++)
+      height += self.getRowHeight(row);
+    return height;
+  };
+
+  self.getSelectedColumnsWidth = function () {
+    const selected = getSelectedContiguousColumns(self.selections, true);
+    if (!selected) return 0;
+    let width = 0;
+    for (let col = selected[0]; col <= selected[1]; col++)
+      width += self.getColumnWidth(self.orders.columns[col]);
+    return width;
+  };
+
+  self.fitSelectedColumns = function (width) {
+    const selectedColumns = self.selections[0];
+    const schema = self.getSchema();
+
+    for (const selectedColumn of selectedColumns) {
+      // Make sure the column is not the row header and that the whole column has in fact been selected.
+      if (selectedColumn >= 0 && self.isColumnSelected(selectedColumn)) {
+        if (isNaN(width)) {
+          const column = schema[self.orders.columns[selectedColumn]];
+          self.fitColumnToValues(column.name);
+        } else {
+          self.sizes.columns[selectedColumn] = width;
+          self.dispatchEvent('resizecolumn', {
```

Comment:
@velitasali , this seems to directly impact your work on #502 — depending on the merge order of both of these PRs, one or both of you may encounter a merge conflict or lost work; perhaps you could both could take a look to see what the impact is?

Reply:
@ndrsn, `fitSelectedColumns` is in `selections/index.js` now. I can rebase my commits if this is merged first. It is up to you :)

Reply:
Ah yeah that sounds like it may be easiest, thanks! Lets wait for @hangxingliu's feedback on my review, and then see if we can merge; it'll probably be tomorrow (Tuesday).

Reply:
@ndrsn @velitasali , I think it is no worry about the conflict on this function. I believe it can be resolved very easy. I have resolved the conflict between this PR and Veli's last PR. And I will modifiy the code in this PR tonight.